addons/point_of_sale/static/src/app/models/related_models/index.js

```diff
@@ -0,0 +1,712 @@
+                            serverData,
+                        });
+                        existingRecord[RAW_SYMBOL] = rawData;
+                        Object.assign(existingRecord, extraFields);
+                        existingRecord.setup(vals);
+                        record = existingRecord;
+                        recordStore.add(record);
+                    } else {
+                        record = this[model]._create(vals, { connectRecords, serverData });
+                    }
+                    if (!(model in results)) {
+                        results[model] = [];
+                    }
+                    results[model].push(record);
+                }
+            }
+            for (const [model, values] of Object.entries(results)) {
+                this[model].triggerEvents("create", {
+                    ids: values.map((v) => v.id),
+                    model: model,
+                });
+            }
+            return results;
+        }
+
+        serializeForORM(record, opts) {
+            return ormSerialization(record, { dynamicModels, ...opts });
+        }
+
+        connect(field, ownerRecord, recordOrIdToConnect) {
+            if (!this[STORE_SYMBOL].hasIndex(field.relation, "id")) {
+                // Not supported model
+                return;
+            }
+            const inverse = inverseMap.get(field);
+            const recordToConnect = this.getRecord(field.relation, recordOrIdToConnect);
+            if (!recordToConnect) {
+                return;
+            }
+
+            if (field.type === "many2one") {
+                const prevConnectedRecordId = ownerRecord[RAW_SYMBOL][field.name];
+                if (prevConnectedRecordId === recordToConnect.id) {
+                    return;
+                }
+                if (!inverse.dummy && inverse.name in recordToConnect) {
+                    this.addItem(recordToConnect, inverse, ownerRecord);
+                }
+                if (prevConnectedRecordId && !inverse.dummy) {
+                    const prevRecord = this[STORE_SYMBOL].getById(
+                        field.relation,
+                        prevConnectedRecordId
+                    );
+                    this.removeItem(prevRecord, inverse, ownerRecord);
+                }
+                ownerRecord[RAW_SYMBOL][field.name] = recordToConnect.id;
+                ownerRecord._markDirty(field.name);
+            } else if (field.type === "one2many") {
+                // It's necessary to remove the previous connected in one2many but it would cause issue for inherited one2many field.
+                // Also, we don't do modification in PoS and we can ignore the removing part to prevent issue.
+                if (!inverse.dummy) {
+                    recordToConnect[RAW_SYMBOL][inverse.name] = ownerRecord.id;
+                    recordToConnect._markDirty(inverse.name);
+                }
+                this.addItem(ownerRecord, field, recordToConnect);
```

Comment:
```suggestion
                                this.connect(inverse, recordToConnect, ownerRecord);
```

maybe you can reuse the already define logic for `one2many`.